README.tpl.md

```diff
@@ -302,6 +294,74 @@ netrc-file = /PATH/TO/.netrc
 GITHUB_TOKEN=$YOUR_GITHUB_TOKEN niv add ...
 ```
 
+### How do I import and use the content of a source?
+
+The way to import a source depend mainly on the content targetted by this
+source. A source could be a file, a repository with no knowledge of nix
+or a repository already in the nix ecosystem.
+
+#### 1. Direct import of a nix based source
+
+In the case of a nix based source, you'll often find a `default.nix`  at the
+root. Let's take this repository as example. We can add it to our `sources.json`
+with the following command.
+
+``` shell
+$ niv add nmattia/niv
+```
+
+We can now import niv to use it a nix expression, e.g.:
+
+``` nix
+{ sources ? import nix/sources.nix }:
+let niv = import sources.niv {};
+in { inherit niv; } # A glorious expression using the reference to niv
+```
+
+#### 2. Import of a nix based source via an overlay
+
+Rather than use the resulting derivation directly, you can add it to your custom 
+nixpkgs via the overlay system.
+
+``` nix
+{ sources ? import nix/sources.nix}:
+let overlay = _: pkgs: {
+      niv = (import sources.niv {}).niv;
+    };
+    nixpkgs = import sources.nixpkgs { overlays = [ overlay ]; config = {}; };
+in { inherit (nixpkgs) niv; } # A glorious expression where nixpkgs.niv is referenced
+```
+
+#### 3. Reference to the source's files in the nix store
+
+You can also reference a simple file, a folder or a repo without nix knowledge
+with niv. In these cases, you can use the source in your nix expression without
+importing it.
+
+The following exemple will compile gnu hello while using this technique to retrieve
+the source. First, we need to add the new source.
+
+``` shell
+$ niv add hello-src -v 2.10 -t 'https://ftp.gnu.org/gnu/hello/hello-<version>.tar.gz'
+```
+
+Then, we can use it inside a nix expression.
+
+``` nix
+{ sources ? import nix/sources.nix }:
+let hello_src = sources.hello-src;
+    nixpkgs =  import sources.nixpkgs {};
+in nixpkgs.stdenv.mkDerivation {
+    pname = "hello";
+    version = "custom";
+    src = hello_src;
+}
+```
+
+:warning: If you have problems, consider using the outPath of the source
```

Comment:
does `:warning:` actually work? is that a GitHub flavored markdown feature?

Reply:
The `:word:` syntax is for writing emoji in github markdow. `:warning:=`:warning: 
It is definitely not a basic markdown feature but other markdown variant integrate this feature too. I'm aware that gitlab's markdown has this feature.

Reply:
hah! of course.